snmp/datadog_checks/snmp/utils.py

```diff
@@ -92,3 +95,69 @@ def oid_pattern_specificity(pattern):
         len(parts),  # Shorter OIDs are less specific than longer OIDs, regardless of their contents.
         parts,  # For same-length OIDs, compare their contents (integer parts).
     )
+
+
+class OIDPrinter(object):
+    """Utility class to display OIDs efficiently.
+
+    This is only meant for debugging, as it makes some assumptions on the data
+    managed, and can be more expensive to use than regular display.
+    """
+
+    def __init__(self, oids, with_value):
```

Comment:
Naming nit (feels like a more natural name for a config option):

```suggestion
    def __init__(self, oids, with_values):
```